File: src/main/java/com/github/pagviewer/editor/PagFileEditorProvider.java
package com.github.pagviewer.editor;

import com.intellij.openapi.fileEditor.FileEditor;
import com.intellij.openapi.fileEditor.FileEditorPolicy;
import com.intellij.openapi.fileEditor.FileEditorProvider;
import com.intellij.openapi.project.DumbAware;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;

import java.util.Locale;

public final class PagFileEditorProvider implements FileEditorProvider, DumbAware {
    @Override
    public boolean accept(@NotNull Project project, @NotNull VirtualFile file) {
        return "pag".equals(file.getExtension() == null ? "" : file.getExtension().toLowerCase(Locale.ROOT));
    }

    @Override
    public @NotNull FileEditor createEditor(@NotNull Project project, @NotNull VirtualFile file) {
        return new PagFileEditor(file);
    }

    @Override
    public @NonNls @NotNull String getEditorTypeId() {
        return "pag-viewer";
    }

    @Override
    public @NotNull FileEditorPolicy getPolicy() {
        return FileEditorPolicy.HIDE_DEFAULT_EDITOR;
    }
}

File: src/main/kotlin/com/github/pagviewer/editor/PagFileEditorProvider.kt
package com.github.pagviewer.editor

import com.intellij.openapi.fileEditor.FileEditor
import com.intellij.openapi.fileEditor.FileEditorPolicy
import com.intellij.openapi.fileEditor.FileEditorProvider
import com.intellij.openapi.project.DumbAware
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile
import java.util.Locale

class PagFileEditorProvider : FileEditorProvider, DumbAware {
    override fun accept(project: Project, file: VirtualFile): Boolean =
        "pag" == (file.extension?.lowercase(Locale.ROOT) ?: "")

    override fun createEditor(project: Project, file: VirtualFile): FileEditor = PagFileEditor(file)

    override fun getEditorTypeId(): String = "pag-viewer"

    override fun getPolicy(): FileEditorPolicy = FileEditorPolicy.HIDE_DEFAULT_EDITOR
}
